test/MatrixFields/scalar_fieldmatrix.jl

```diff
@@ -0,0 +1,218 @@
+using Test
+using JET
+
+import ClimaCore: Geometry, Domains, Meshes, Spaces, Fields, MatrixFields
+import ClimaCore.Utilities: half
+import ClimaComms
+import ClimaCore.MatrixFields: @name
+ClimaComms.@import_required_backends
+include("matrix_field_test_utils.jl")
+
+@testset "get_field_first_index_offset" begin
+    struct Singleton{T}
+        x::T
+    end
+    struct Two_fields{T1, T2}
```

Comment:
```suggestion
    struct TwoFields{T1, T2}
```
We typically use camel case for type names